app/scripts/modules/core/src/api/ApiService.ts

```diff
@@ -1,193 +1,254 @@
+ */
+class AngularJSHttpBackend implements IHttpClientBackend {
+  delete = <T = any>(requestConfig: IRequestBuilderConfig) => this.request<T>('DELETE', requestConfig);
+  get = <T = any>(requestConfig: IRequestBuilderConfig) => this.request<T>('GET', requestConfig);
+  post = <T = any>(requestConfig: IRequestBuilderConfig) => this.request<T>('POST', requestConfig);
+  put = <T = any>(requestConfig: IRequestBuilderConfig) => this.request<T>('PUT', requestConfig);
+
+  private request<T>(method: 'GET' | 'POST' | 'PUT' | 'DELETE', requestConfig: IRequestBuilderConfig): PromiseLike<T> {
+    return $http<T>({ ...requestConfig, method }).then((response) => {
+      const contentType = response.headers('content-type');
 
-  public static readonly invalidContentMessage = 'API response was neither JSON nor zero-length html or text';
-
-  private static getData(result: any): PromiseLike<any> {
-    return $q((resolve, reject) => {
-      const contentType = result.headers('content-type');
       if (contentType) {
-        const isJson = contentType.match(/application\/(.+\+)?json/); // e.g application/json, application/hal+json
+        // e.g application/json, application/hal+json
+        const isJson = contentType.match(/application\/(.+\+)?json/);
         // e.g. application/yaml, application/x-yaml; it's regex, let's not get too fancy
         const isYaml = contentType.match(/application\/(.+-)?yaml/);
-        const isZeroLengthHtml = contentType.includes('text/html') && result.data === '';
-        const isZeroLengthText = contentType.includes('text/plain') && result.data === '';
+        const isZeroLengthHtml = contentType.includes('text/html') && (response as any).data === '';
+        const isZeroLengthText = contentType.includes('text/plain') && (response as any).data === '';
         if (!(isJson || isYaml || isZeroLengthHtml || isZeroLengthText)) {
           AuthenticationInitializer.reauthenticateUser();
-          reject(new InvalidAPIResponse(API.invalidContentMessage, result));
+          throw new InvalidAPIResponse(invalidContentMessage, response);
         }
       }
 
-      return resolve(result.data);
+      return response.data;
     });
   }
+}
 
-  private static internalOne(config: IRequestConfig): (...urls: string[]) => IRequestBuilder {
-    return (...urls: string[]) => {
-      urls.forEach((url: string) => {
-        if (url) {
-          config.url = `${config.url}/${url}`;
-        }
-      });
-
-      return this.baseReturn(config);
-    };
-  }
+function joinPaths(...paths: IPrimitive[]) {
+  // coerce paths toString() in case somebody sends in a url object
+  // according to https://github.com/spinnaker/deck/pull/6927
+  return paths
+    .filter((path) => !isNil(path) && path !== '')
+    .map((path) => path.toString())
+    .map((path) => path.replace(/^\/+/, '')) // strip leading slashes
+    .map((path) => path.replace(/\/+$/, '')) // strip trailing slashes
+    .join('/');
+}
```

Comment:
A naive question, but where is the encoding happening? I was kind of expecting it to happen here.

Reply:
Good question.  It's not merged yet, but it will go in the `.path()` method